### PR TITLE
add exports to package.json + build icons esm file

### PR DESCRIPTION
**Pull request by eran-cohen**
https://monday.monday.com/boards/245345663/views/62248892/pulses/2970462587

using [exports](https://webpack.js.org/guides/package-exports/) in package json to specify path for icons pointing to a newly created dist/icons/index.js file listing all available icons to support tree shaking.

We can now do `import { Close } from 'monday-ui-react-core/icons';` and tree shaking works fine.  
CommonJs and /dist path mapping still works for backward compatibility but I think we should add a dageroid rule discouraging linking directly into the dist folder

### CODE REVIEW

**Review by sahariko — APPROVED** [review]
Amazing stuff!

LGTM but let's wait for a review from someone from the Vibe team.

**Review by orrgottlieb — APPROVED** [review]

**Pull request merged**
